Can't have multiple references to arrays in `ConfigureInterop`

I can confirm that #38 allows for support for arrays in interop, but if it seems that if there is more than one reference to an array type of a custom script, it prints these errors to the console: 

```
ReinteropInitializer.cs(1834,37): error CS0102: The type 'ReinteropInitializer' already contains a definition for 'CesiumForUnity__Property_get_LengthType' [C:\Dev\cesium-unity\CesiumForUnity\CesiumForUnity.csproj]
ReinteropInitializer.cs(1835,80): error CS0102: The type 'ReinteropInitializer' already contains a definition for 'CesiumForUnity__Property_get_LengthDelegate' [C:\Dev\cesium-unity\CesiumForUnity\CesiumForUnity.csproj]
ReinteropInitializer.cs(1837,35): error CS0111: Type 'ReinteropInitializer' already defines a member called 'CesiumForUnity__Property_get_Length' with the same parameter types [C:\Dev\cesium-unity\CesiumForUnity\CesiumForUnity.csproj]
```

To reproduce: add this line to `ConfigureInterop.cs`:
```
CesiumGeoreference[] georeferences = UnityEngine.Object.FindObjectsOfType<CesiumGeoreference>();
```

Since there is a line containing `CesiumRasterOverlay[]` in the file, it will print these errors. This works even without the variable assignment, just the line `UnityEngine.Object.FindObjectsOfType<CesiumGeoreference>();` Commenting out the lines related to `CesiumRasterOverlay[]` will cause the interop to run successfully.

--- a/CesiumForUnity/ConfigureReinterop.cs
+++ b/CesiumForUnity/ConfigureReinterop.cs
@@ -194,5 +194,7 @@ internal partial class ConfigureReinterop
         {
             GameObject goFromArray = gos[i];
         }
+
+        CesiumGeoreference[] georeferences = UnityEngine.Object.FindObjectsOfType<CesiumGeoreference>();
     }
 }

--- a/Reinterop/CppType.cs
+++ b/Reinterop/CppType.cs
@@ -206,8 +206,8 @@ namespace Reinterop
 
         public void AddForwardDeclarationsToSet(ISet<string> forwardDeclarations)
         {
-            // Primitives do not need to be forward declared
-            if (Kind == InteropTypeKind.Primitive)
+            // Primitives and generic parameters do not need to be forward declared
+            if (Kind == InteropTypeKind.Primitive || Kind == InteropTypeKind.GenericParameter)
                 return;
 
             string template = "";

--- a/Reinterop/Interop.cs
+++ b/Reinterop/Interop.cs
@@ -254,6 +254,7 @@ namespace Reinterop
 
         public static string GetUniqueNameForType(CSharpType type)
         {
+            string name = type.Symbol.Name;
             string genericTypeHash = "";
             INamedTypeSymbol? named = type.Symbol as INamedTypeSymbol;
             if (named != null && named.IsGenericType)
@@ -261,7 +262,14 @@ namespace Reinterop
                 genericTypeHash = HashParameters(null, named.TypeArguments);
             }
 
-            return $"{type.GetFullyQualifiedNamespace().Replace(".", "_")}_{type.Symbol.Name}{genericTypeHash}";
+            IArrayTypeSymbol? arraySymbol = type.Symbol as IArrayTypeSymbol;
+            if (arraySymbol != null)
+            {
+                name = "Array1";
+                genericTypeHash = HashParameters(null, new[] { arraySymbol.ElementType });
+            }
+
+            return $"{type.GetFullyQualifiedNamespace().Replace(".", "_")}_{name}{genericTypeHash}";
         }
 
         public static void GenerateForType(CppGenerationContext context, TypeToGenerate item, GeneratedResult result)

--- a/Reinterop/Methods.cs
+++ b/Reinterop/Methods.cs
@@ -68,7 +68,7 @@ namespace Reinterop
                             template <{{string.Join(", ", method.TypeParameters.Select(parameter => "typename " + parameter.Name))}}>
                             {{modifiers}}{{genericReturn.GetFullyQualifiedName()}} {{method.Name}}({{genericParametersString}}){{afterModifiers}};
                             """,
-                        TypeDeclarationsReferenced: genericMethod.Parameters.Select(p => CppType.FromCSharp(context, p.Type))
+                        TypeDeclarationsReferenced: new[] { genericReturn }.Concat(genericMethod.Parameters.Select(p => CppType.FromCSharp(context, p.Type)))
                         ));
                 }
 

